client/src/actions/team.js

```diff
@@ -1,15 +1,70 @@
 /**
  * This file houses our team-related Action Creators.
  */
-import * as api from '../api';
+import * as api from "../api";
 import {
-    LINK_TEAM_TWITTER, 
-    UNLINK_TEAM_TWITTER, 
-    FETCH_TEAM_INFO,     
-    GET_TEAM_MEMBERS_BY_TEAM_ID, 
-    CREATE_TEAM_MEMBER, 
-    UPDATE_TEAM_MEMBER, 
-    DELETE_TEAM_MEMBER } from './types';
+  LINK_TEAM_TWITTER,
+  UNLINK_TEAM_TWITTER,
+  FETCH_TEAM_INFO,
+  GET_TEAM_MEMBERS_BY_TEAM_ID,
+  CREATE_TEAM_MEMBER,
+  UPDATE_TEAM_MEMBER,
+  DELETE_TEAM_MEMBER,
+  ADD_TEAM,
+} from "./types";
+
+/**
+ * Adds a new team to redux store and database. If api request fails the team state's requestFailed attribute will be 'true'
```

Comment:
A suggestion cause scholar did it for the imported publications, you could have a ADD_TEAM_FAIL and ADD_TEAM_SUCCESS dispatch action to differentiate between whether the api request fails or not?

Reply:
Yep I did initially get inspired by Scholar and tried do it that way. However, the difference is that Scholar used a different React component for the error, while in this case the error is displayed by the same component. Happy to go into further discussion on how I wasn't able to use it in my case, but better to do it over a call :)

Reply:
I'm going to try a different method to having a callback tho, let me try that